### PR TITLE
make output type switchable

### PR DESCRIPTION
**Pull request by baronfel**
## Proposed Changes

F# slack has expressed a desire to make an executable from this template. 

## Types of changes

What types of changes does your code introduce to MiniScaffold?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] Build and tests pass locally
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added necessary documentation (if appropriate)

## Further comments

I would _love_ to test this but I can't figure out how to uninstall a previous version of the template.
Also the build script seems to be building the inner Content folder, which with this change is no longer buildable, because the output type `MyOutputType` isn't valid.  I didn't see a good way to keep that a buildable value but not accidentally find-and-replace a billion instances of `Lib` in the project.

### CODE REVIEW

**Comment by TheAngryByrd** [comment]
So the [giraffe-template build](https://github.com/giraffe-fsharp/giraffe-template/blob/master/build.ps1)  has some permutation tests we could probably imitate with FAKE

File: Content/.template.config/template.json [hunk 1]
```diff
@@ -1,4 +1,5 @@
 {
+  "$schema": "http://json.schemastore.org/template",
```
Comment (TheAngryByrd) [inline Content/.template.config/template.json]: Ooo I didn't know about this!

**Comment by TheAngryByrd** [comment]
One of the other issues is that `OutputType` doesn't directly map to `TargetFrameworks` anymore

Libs are
netstandard1.6
netstandard2.0
net461

Exe are
netcoreapp1.1
netcoreapp2.0
net461

**Comment by TheAngryByrd** [comment]
Also from slack discussion, we probably need a `DotnetPublish` target.

https://docs.microsoft.com/en-us/dotnet/core/deploying/deploy-with-cli

Probably  projects filtered by  src and `OutputType = exe` and a crossproduct of `RuntimeIdentifiers` and `TargetFrameworks`.  Then call dotnet publish on those projects

**Comment by ScottHutchinson** [comment]
Although uninstall with -u does not work, re-installing (as shown below) *does* update the template to the latest version on nuget. Not sure how how you test a branch though.

dotnet new -i "MiniScaffold::*"

**Comment by TheAngryByrd** [comment]
You should be able to uninstall with `dotnet new -u MiniScaffold`.

As for testing a branch, you'll have to build it yourself and install the nupkg that resides in the `./dist` folder so something like `dotnet new -i ./dist/MiniScaffold.0.6.1.nupkg`.

**Comment by TheAngryByrd** [comment]
Also looks like we'll need to add a Program.fs or corresponding some entrypoint

**Comment by TheAngryByrd** [comment]
I'm thinking there's enough different here that we'll probably want to create another folder under `Content` and scaffold up another similar project structure but with things like [Argu](https://fsprojects.github.io/Argu/) and [dotnet-packaging](https://github.com/qmfrederik/dotnet-packaging)

**Comment by TheAngryByrd** [comment]
Going to close in favor of #105

**Pull request closed without merging**